src/Zenaton/Exceptions/ApiException.php

```diff
@@ -18,4 +18,33 @@ public static function unexpectedStatusCode($code)
     {
         return new static("The API responded with an unexpected status code: {$code}.");
     }
+
+    /**
+     * Creates a new instance of the exception when a connection error was received.
+     *
+     * @return self
+     */
+    public static function connectionError(\Exception $previous)
+    {
+        return new static("A connection error occurred while trying to send a request to the Zenaton API: {$previous->getMessage()}.", 0, $previous);
+    }
+
+    /**
+     * Creates a new instance of the exception when an error list is received.
+     *
+     * @return self
+     */
+    public static function fromErrorList(array $errors)
+    {
+        $errorMessages = implode("\n", array_map(static function ($error) {
+            return '  - '.$error->message;
+        }, $errors));
+
+        $message = <<<MESSAGE
+The Zenaton API returned some errors:
```

Comment:
In Ruby we get `<<-` and `<<~` for heredocs. The later allows you to keep a consistent indentation.
This
```ruby
class MyClass
  def my_method
    <<-DOC
My heredoc message
    DOC
  end
end
```
becomes this
```ruby
class MyClass
  def my_method
    <<~DOC
      My heredoc message
    DOC
  end
end
```
Are there no equivalent in PHP?

Reply:
There is since PHP 7.3 but since we support versions as far as PHP 5.6 this is not an option, unfortunately.